server/eclipse-project/src/main/java/us/freeandfair/corla/csv/DominionCVRExportParser.java

```diff
@@ -406,6 +409,66 @@ private void checkForFlush() {
   }
   
   /**
+   * Commits the currently outstanding CVRs and updates the county dashboard
+   * accordingly.
+   */
+  private void commitCVRsAndUpdateCountyDashboard() {
+    // commit all the CVR records and contest tracking data
+    Persistence.commitTransaction();
+    
+    boolean success = false;
+    int retries = 0;
+    while (!success && retries < UPDATE_RETRIES) {
+      try {
+        retries = retries + 1;
+        Main.LOGGER.debug("updating county " + my_county.id() + " dashboard, attempt " +
+                          retries);
+        Persistence.beginTransaction();
+        final CountyDashboard cdb = 
+            Persistence.getByID(my_county.id(), CountyDashboard.class);
+        // if we can't get a reference to the county dashboard, we've got problems -
+        // but we'll deal with them elsewhere
+        if (cdb == null) {
+          Persistence.rollbackTransaction();
+        } else {
+          cdb.setCVRsImported(my_record_count);
+          Persistence.saveOrUpdate(cdb);
+          Persistence.commitTransaction();
+          success = true;
+        }
+      } catch (final PersistenceException e) {
+        // something went wrong, let's try again
+        if (Persistence.canTransactionRollback()) {
+          try {
+            Persistence.rollbackTransaction();
+          } catch (final PersistenceException ex) {
+            // not much we can do about it
+          }
+        }
+        // let's give other transactions time to breathe
+        try {
+          final long delay = 
+              ExponentialBackoffHelper.exponentialBackoff(retries, TRANSACTION_SLEEP_MSEC);
+          Main.LOGGER.info("retrying county " + my_county.id() + 
+                           " dashboard update in " + delay + "ms");
+          Thread.sleep(delay);         
+        } catch (final InterruptedException ex) {
+          // it's OK to be interrupted
+        }
+      }
+    }
+    // we always need a running transaction
+    Persistence.beginTransaction();
+    if (success && retries > 1) {
+      Main.LOGGER.info("updated state machine for county " + my_county.id() + 
+                       " in " + retries + " tries");
+    } else if (!success) {
+      throw new PersistenceException("could not update state machine for county " + 
+                                     my_county.id() + " after " + (retries - 1) + " tries");
```

Comment:
Should we really subtract 1 from `retries` here?

Reply:
No. I thought I’d caught all of those. I will fix that before merging.